### ISSUE TITLE
Document keyboard shortcuts

### ISSUE THREAD

**Issue by mrocklin** [jupyterlab/jupyterlab#2976]
So my original question was

*Is there a keyboard shortcut to navigate between panes in JupyterLab?*

So I went to [the documentation](http://jupyterlab-tutorial.readthedocs.io) and searched for shortcuts and the only link was the "Extensions Developer Guide", a link scary enough to make me turn tail and run :)

Oddly I find that the quick access to the command palette is one of the things that keeps me coming back to JLab.  I suspect that there are other shortcuts that I would enjoy knowing about.  Are these listed someplace?

**Comment by jasongrout** [comment]
Go to the File | Settings menu option to see the full list of keyboard shortcuts, and customize them as well.

**Comment by Juanlu001** [comment]
For reference: to navigate between tabs, customize these two actions in the settings editor as @jasongrout said:

https://github.com/jupyterlab/jupyterlab/blob/7c9a56df/packages/application-extension/src/index.ts#L24-L28

**Comment by willingc** [comment]
@jasongrout Is there a way to easily render the Keyboard Shortcuts in File | Settings? I would like a cheatsheet. :smile:

**Comment by jasongrout** [comment]
Great idea! Presumably the "Table view" would do something like that, but it doesn't. To work around it, you could copy the json and write a short program to render the shortcuts in a table. However, we should definitely support something like a nice view you could print.

**Comment by sccolbert** [comment]
Yep. We need to give the settings UI a bit of UX love.

**Comment by fperez** [comment]
In addition to rendering the raw shortcuts data structure (great idea, but it's likely to be large and a bit hard to follow), would it be useful to offer a menu entry like we have in classic? This page:

![image](https://user-images.githubusercontent.com/57394/30681762-c02f2b66-9e5b-11e7-874a-fa1ac1bae39a.png)

is one of the most useful ones for me and one of the first things I teach folks so they can find their way around the system...  Having something like that, listing potentially only a subset of more commonly used shortcuts, and that is somewhat thematically organized, can be very valuable.

**Comment by jzf2101** [comment]
This should probably go in the "Getting Started" section

**Comment by afshin** [comment]
@jzf2101 I think that describing the current state of affairs for keyboard shortcuts in the documentation here: https://github.com/jupyterlab/jupyterlab/blob/master/docs/source/user/interface.rst#context-menus right after the "context menus" section might be another place where it makes sense.

**Comment by jzf2101** [comment]
Ok, I can write it explictily there and link to it

**Comment by afshin** [comment]
Good plan 👍

**Comment by afshin** [comment]
![shortcuts](https://user-images.githubusercontent.com/159529/35583482-fe23ed62-05e9-11e8-8e64-e4ac438f024e.gif)

That is where you can discover shortcuts, currently, before a shortcuts UI is written.

**Comment by jzf2101** [comment]
Sweet thanks

**Comment by jzf2101** [comment]
@ian-r-rose how can we turn that into a video?

**Comment by afshin** [comment]
![vim-emacs](https://user-images.githubusercontent.com/159529/35583652-935ed2fc-05ea-11e8-9657-b10a142b1257.gif)
And this is the vim/emacs mode.

**Comment by saulshanabrook** [comment]
FYI https://github.com/jupyterlab/jupyterlab/issues/4004 is still open and tracks adding a menu item itself, like in the classic notebook. It wasn't clear to me if we decided conclusively here not to have the item.

**Comment by lstodd** [comment]
Do we know if there is a plan to add a keyboard shortcut link yet? I've been using the shortcuts in settings as helfully described above (thanks @afshin!) but would be good to have a shortcut to the shortcuts!  :)

**Comment by afshin** [comment]
The plan is to build an actual shortcuts editor before the 1.0 release. But before then, I think @saulshanabrook is right about linking to the setting editor opened up to the shortcuts.

**Comment by jasongrout** [comment]
@ellisonbg - a shortcuts editor might be a good student project.

**Comment by fonnesbeck** [comment]
Is there not a shortcuts cheat sheet like the one Fernando referenced from the notebooks? Its not easy to browse through the shortcuts editor to find what you need.

**Comment by willingc** [comment]
I agree with @fonnesbeck that it would be lovely to have a cheatsheet in the docs that could be printed out.

**Comment by bollwyvl** [comment]
Did a talk the other night with lab and got two questions about keyboard
shortcuts. So I made this extension:

https://github.com/deathbeds/jupyterlab-keycaps

It's basically a cut down command pallette to just those commands that have
shortcuts.

Thoughts?

**Comment by breckuh** [comment]
I just wrote a bit of code to bring back the shortcuts modal. Anyone know if this is something already in the works?

Basic working code here: https://github.com/breckuh/jupyterlab/commit/dfdc6f2220a93c954a8936051f5c7cb3151d6935

Here's what it looks like now (pre-styling/design):

![demo](https://user-images.githubusercontent.com/37387936/45334000-8929ee00-b514-11e8-8313-e38e3c68092a.gif)

**Comment by saulshanabrook** [comment]
@breckuh There is this GUI editor now the shortcuts: https://github.com/jupyterlab/jupyterlab-shortcutui

**Comment by breckuh** [comment]
Looks good @saulshanabrook !

**Comment by louis925** [comment]
> ![shortcuts](https://user-images.githubusercontent.com/159529/35583482-fe23ed62-05e9-11e8-8e64-e4ac438f024e.gif)
> 
> That is where you can discover shortcuts, currently, before a shortcuts UI is written.

Is there a documentations for this. What is the Accel key? And, where are the Command, and Option for mac, or Alt key for Windows? Should I use Delete or Del? Command or Cmd? Backspace?

**Comment by Jakeler** [comment]
I made a shortcut overview (just static, but more compact) that can be generated from the config, if anyone is interested: https://blog.ja-ke.tech/2019/01/20/jupyterlab-shortcuts.html
or https://github.com/Jakeler/jupyter-shortcuts.

The Accel key is Ctrl at least on Linux/KDE @louis925. Otherwise you have to try it, i guess.

**Comment by saulshanabrook** [comment]
@jakeler wow that looks really nice!

I wonder if we should add this to core...

**Comment by bollwyvl** [comment]
Yeah those look really good!

+1 to a contextually filtered shortcut list in core. -100 to any more
modals, anywhere, so as a sidebar or dock panel item, please.

Having a printable cheat sheet of YOUR lab is also a fabulous play.

Here's what we came up with in keycaps:

https://user-images.githubusercontent.com/45380/43494634-d233f218-9501-11e8-90de-0f7c817c3357.gif
